<launch>
    <node name="map_server" pkg="map_server" type="map_server" args="$(find stage_simulation)/config/map_8thfloor.yaml">
        <param name="frame_id" value="/map" />
    </node>
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find stage_simulation)/config/simulation.rviz"/>    

    <include file="$(find stage_simulation)/launch/stage.launch"/>
    <include file="$(find stage_simulation)/launch/amcl.launch" />
       
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        <rosparam file="$(find stage_simulation)/config/costmap_common_params_stage.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find stage_simulation)/config/costmap_common_params_stage.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find stage_simulation)/config/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find stage_simulation)/config/global_costmap_params.yaml" command="load" />
        <rosparam file="$(find stage_simulation)/config/base_local_planner_params.yaml" command="load" />
        <!--rosparam file="$(find stage_simulation)/config/dwa_local_planner_params.yaml" command="load" /-->
        <param name="base_global_planner" type="string" value="active_perception_interface/ActivePerceptionInterface" />
	<!--param name="base_local_planner" type="string" value="dwa_local_planner/DWAPlannerROS" /-->
        <param name="controller_frequency" type="double" value="20.0" />
        <rosparam param="recovery_behaviors">[{name: conservative_reset, type: clear_costmap_recovery/ClearCostmapRecovery},
                                              {name: aggressive_reset, type: clear_costmap_recovery/ClearCostmapRecovery}]</rosparam>
        <param name="planner_patience" type="double" value="30.0"/>
        <param name="controller_patience" type="double" value="60.0"/>
        <remap from="map" to="/map"/>
    </node>
    
    <node pkg="active_perception_controller" type="motion_planner.py" name="entropy_motion_planner" output="screen">
	    <param name="rrt_nearest_part_bias" type="double" value="0.2" />
	    <param name="rrt_total_dist_bias" type="double" value="0.05" />
	    <param name="rrt_entropy_bias" type="double" value="10" />
	    <param name="max_rrt_iterations" type="int" value="500" />
	    <param name="max_path_size" type="int" value="20" />	
	    <param name="rrt_eta" type="double" value="0.5" />
    </node>  

    <node pkg="active_perception_controller" type="experiment_builder.py" name="experiment_builder" output="screen">
	    <param name="number_of_goals" type="int" value="5" />
	    <param name="number_of_people" type="int" value="2" />
	    <param name="checkpoints_per_person" type="int" value="2" />
	    <param name="experiment_name" type="string" value="workshop_data" />
    </node>  
</launch>
